metadata_backend/database/db_service.py

```diff
@@ -95,29 +96,46 @@ async def update(self, collection: str, accession_id: str,
         updated to object, can replace previous fields and add new ones.
         :returns: True if operation was successful
         """
-        find_by_id_query = {"accessionId": accession_id}
-        update_operation = {"$set": data_to_be_updated}
-        result = await self.database[collection].update_one(find_by_id_query,
-                                                            update_operation)
-        LOG.debug(f"DB doc updated for {accession_id}.")
-        return result.acknowledged
+        find_by_id = {"accessionId": accession_id}
+        update_op = {"$set": data_to_be_updated}
+        old_data = await self.database[collection].find_one(find_by_id)
+        if not old_data:
+            reason = f"Object with accession id {accession_id} was not found."
+            LOG.error(reason)
+            raise web.HTTPNotFound(reason=reason)
```

Comment:
For consistency, I would say its good to have all aiohttp related logic in operators module, and keep this module purely db related. Maybe return something thats telling to raise errors on operators?

Reply:
Open for opposing views as well :)

Reply:
we need to raise an error for data not found, what error would be raised here if not one for  Http Not found ?

Reply:
did not see such error in `motor`. I assume in can raise a general one/ custom one and then catch it in operators, but feels like a lot of code when in can just raise it here .

Reply:
I mean that all the other http errors are raised inside api-packages modules (handlers, operators, middlewares). We're breaking that logic by raising errors here in db_service module

Reply:
ok, i see then we need a function in db to check there is data and based on that raise an error on operators before doing anything ... ?

Reply:
i was thinking this was easy to change, but it involves more logic to change as the operator is expecting true / false / None and errors with HTTPBadRequest if it gets anything else than true, which is not correct - what I am saying is either we raise or return something it is still a trade off to make.

if i implement a function to check data is there, while it works for replace, update, then comes the use of `deleted_count` which I can make use of and not do another db call to find if data exists.

anyway will try to figure something without a lot of changes

Reply:
If my suggestion is bigger task to do / overcomplicating things, we can do this like you originally did

Reply:
I mean, let's not make things harder on purpose

Reply:
refactored, hope now it is ok.